packages/babel-traverse/src/path/inference/index.ts

```diff
@@ -23,12 +23,15 @@ import type * as t from "@babel/types";
  * Infer the type of the current `NodePath`.
  */
 
-export function getTypeAnnotation(): t.FlowType {
-  if (this.typeAnnotation) return this.typeAnnotation;
-
-  let type = this._getTypeAnnotation() || anyTypeAnnotation();
+export function getTypeAnnotation(this: NodePath): t.FlowType {
+  let type = this.getData("typeAnnotation");
```

Comment:
With this code change we use the internal map storage that `NodePath` supports, rather than just adding an extra property on the `path` object.